spread.yaml

```diff
@@ -1060,12 +1060,36 @@ suites:
         environment:
             CONNECTIONS_PERCENTAGE: '$(HOST: echo "${PERF_CONNECTIONS_PERCENTAGE:-100}")'
             DISCONNECT_INTERFACES: '$(HOST: echo "${PERF_DISCONNECT_INTERFACES:-true}")'
+            CPU_LOAD: '$(HOST: echo "${PERF_CPU_LOAD:-}")'
+            LOAD_DURATION: '$(HOST: echo "${PERF_LOAD_DURATION:-1200}")'
+            NUM_PARALLEL: '$(HOST: echo "${PERF_NUM_PARALLEL:-10}")'
         manual: true
         prepare: |
             "$TESTSLIB"/prepare-restore.sh --prepare-suite
         prepare-each: |
             "$TESTSLIB"/prepare-restore.sh --prepare-suite-each
+            if [ -n "$CPU_LOAD" ]; then
+                # Start load generator and wait until it is active
+                # It is generated as a systemd service to make it independent of snapd
+                # and also to make sure it is not blocking spread scripts
+                snap install --edge --devmode test-snapd-load-generator
+                tests.systemd create-and-start-unit load-generator "/snap/bin/test-snapd-load-generator.run -d $LOAD_DURATION -l $CPU_LOAD"
+                tests.systemd wait-for-service -n 5 --wait 1 --state active load-generator
+            fi
         restore-each: |
+            if [ -n "$CPU_LOAD" ]; then
+                # Now we make sure all the load generator processes are killed before
+                # the load-generator service is removed
+                #shellcheck disable=SC2009
```

Comment:
Leftover shellchecks (below as well)? Should these be enabled, or removed, or put behind some variable?